### ISSUE TITLE
Iterable.pickOne and Iterable.pickSome throws ArgumentError for optional Random parameter

### ISSUE THREAD

**Issue by bierbaumtim** [felixblaschke/supercharged#19]
Thanks for this awesome package. 

I found one small issues:

The example shows the usage of pickOne and pickSome as follows:
```
[1, 2, 3].pickOne();   // Get a random item
[1, 2, 3].pickSome(2); // or multiple random items
```
But this is not possible, because both methods contains a `ArgumentError.checkNotNull` for the optional random parameter.
Therefore the random parameter is currently required, which can confuse someone who is not looking into the package code.

It would be nice if both methods internally creates an instance of Random if no instance is provided.

**Comment by felixblaschke** [comment]
Thank for reporting. You are right. `random` is an optional parameter and needs no null check.

**Comment by felixblaschke** [comment]
I resolved this issue in supercharged `1.11.1` and supercharged_dart `1.3.1`.

May you also can confirm that it works as intended? :-)

**Comment by bierbaumtim** [comment]
Works perfectly.

Thanks for the fast fix.
